CU-25 Derecho al olvido

angbongon opened the pull request:
Se ha implementado de forma que se eliminan todos los datos personales del usuario, pero los eventos creados por el mismo se mantienen, al igual que las puntuaciones (en las que es revisor) y las transacciones en las que ha intervenido.

Commit:
[WIP] Migraciones

Commit:
[feat] Eliminación de datos de usuario

angelcalzadollamas reviewed (changes requested):
Cosas que se deben corregir:

- Si busco un evento del usuario delete que no ha acabado, me puedo inscribir a él (mi consejo: poner los eventos de los usuarios eliminados a pasado)
- Si el usuario delete tiene un enrollment de un evento aceptado y me logeo como el anfitrión de ese evento, puedo ver que un tal "delete" va a asistir a mi evento . (mi consejo, cancelar los enrollments de eventos aún no empezados)

Commit:
[feat][fix] Eventos futuros se eliminan junto con sus enrollments

También se controla que no se pueda introducir el token del usuario
eliminado en el registro de usuario

angelcalzadollamas reviewed (changes requested):
Cosas que corregir:

1. Si un usuario A es un asistente aceptado de un evento pasado de un usuario B. Cuando B borra sus datos, A puede puntuar al usuario Delete. Esto es porque al ser el evento pasado, no borramos ni el evento ni sus enrollments por lo que sigue saliendo en la lista de eventos como asistente de A. Posible solución: Eliminar los enrollments a los eventos del usuario delete.

2. Debes eliminar los enrollments que ha hecho deleted a eventos futuros también porque estos cuando acaban, si pongo la id de deleted en la puntuación, lo puedo puntuar.

Resumen: Creo que esto alfinal va a tener brechas en más de un sitio y además es muy complicado de revisar, yo lo dejaría en guardar solo los eventos pasados del usuario y eliminar todos lo demas: Enrollments a esos eventos, ratings al usuario eliminado, enrollments que el eliminado ha hecho, ratings que el eliminado ha hecho, etc.

Te dejo un ejemplo para que veas como soy capaz de puntuar al deleted:
![Puedo puntuar a deleted](https://user-images.githubusercontent.com/28622424/78463653-bc783300-76df-11ea-919a-03163c40d5fb.PNG)

Commit:
[fix] Correcciones de incoherencias

Se eliminan los enrollments a eventos futuros del usuario eliminado,
junto con esto se controla que no se puedan hacer ratings nuevos sobre
el usuario anonimizado.

Commit:
[fix] Se evitan nuevos ratings sobre usuario deleted.

angelcalzadollamas reviewed (changes requested):
Correcciones:

1. Cuando un usuario va a puntuar a los asistentes de un evento y un asistente se eliminó sale en la lista de asistentes como deleted. Te dejo una imagen para que lo veas.

![Huespedes del evento](https://user-images.githubusercontent.com/28622424/78510438-5ea62280-7795-11ea-9a25-68d353c088b8.PNG)


2. Cuando un usuario intenta puntuar al anfitrión delete le sale dos mensajes de que no puede, creo que se solapan. Te dejo una imagen abajo con lo que me refiero. Es la primera fila

![Veo que se solapan estos mensajes de no disponible](https://user-images.githubusercontent.com/28622424/78510422-5221ca00-7795-11ea-9d0c-123ee3e199a5.PNG)

angbongon commented:
> Correcciones:
> 
> 1. Cuando un usuario va a puntuar a los asistentes de un evento y un asistente se eliminó sale en la lista de asistentes como deleted. Te dejo una imagen para que lo veas.
> 
> ![Huespedes del evento](https://user-images.githubusercontent.com/28622424/78510438-5ea62280-7795-11ea-9a25-68d353c088b8.PNG)
> 
> 
> 2. Cuando un usuario intenta puntuar al anfitrión delete le sale dos mensajes de que no puede, creo que se solapan. Te dejo una imagen abajo con lo que me refiero. Es la primera fila
> 
> ![Veo que se solapan estos mensajes de no disponible](https://user-images.githubusercontent.com/28622424/78510422-5221ca00-7795-11ea-9d0c-123ee3e199a5.PNG)

La segunda foto es algo relacionado más bien con el front. La primera es porque en principio los enrollments hechos a eventos pasados no se eliminan, si quieres que aparezca otra clase de información es algo referente al front, si quieres que se eliminen todos los enrollments del usuario eliminado lo hago así.

angbongon commented:
Yo creo que para el asistente lo mejor es dejar no disponible en todos los campos, pero como prefieras

Commit:
[fix] Se eliminan todos los enrollments

angbongon commented:
La parte de front no la voy a tocar porque tampoco sé cómo lo quieren poner @angcallla

angelcalzadollamas commented:
Vale, me lo he apuntado para que lo hablemos con los front.

angelcalzadollamas reviewed (dismissed):
Todo correcto, a la espera de arreglar el solapamiento front end cuando se acepten la otra pull request

angbongon committed:
Merge branch 'dev' into feat/be/CU-25
